S/SCS_MKL/build_tarballs.jl

```diff
@@ -0,0 +1,44 @@
+using BinaryBuilder
+
+name = "SCS_MKL"
+version = v"3.2.1"
+
+# Collection of sources required to build SCSBuilder
+sources = [
+    GitSource("https://github.com/cvxgrp/scs.git", "c785d2fad46a30f1d43764d682509d0b56e5c64f")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir/scs*
+scsflags="DLONG=1 USE_OPENMP=0 BLAS32=1 NOBLASSUFFIX=1"
+mklflags="-L${prefix}/lib -Wl,--no-as-needed -lmkl_rt -lmkl_gnu_thread -lmkl_core -lpthread -lm -ldl"
```

Comment:
Not sure `-Wl,--no-as-needed` is needed (see what I did there)

Reply:
`-Wl, --no-as-needed` is suggested by intel: https://www.intel.com/content/www/us/en/developer/tools/oneapi/onemkl-link-line-advisor.html and therefore is also here:
https://github.com/cvxgrp/scs/blob/c785d2fad46a30f1d43764d682509d0b56e5c64f/scs.mk#L136
I don't feel competent to change any of those ;)